cpp/src/neighbors/ivf_flat/ivf_flat_interleaved_scan.cuh

```diff
@@ -134,6 +137,10 @@ struct loadAndComputeDist {
 #pragma unroll
       for (int k = 0; k < Veclen; ++k) {
         compute_dist(dist, queryRegs[k], encV[k]);
+        if constexpr (ComputeNorm) {
+          norm_query += queryRegs[k] * queryRegs[k];
+          norm_data += encV[k] * encV[k];
+        }
```

Comment:
We should ask the question whether it is more efficient to compute the norms here or precompute them: 
- we could store norm of database vectors in the index
- query norm is recalculated every time we process vector in the cluster. I expect that it would be better to use precalculated query norm, but we should confirm that by measurement. Please open an issue to track this, and improve this in a follow-up PR.